doc/man1/openssl-cmp.pod.in

```diff
@@ -0,0 +1,1142 @@
+
+=back
+
+
+=head2 Mock server options, for testing purposes only
+
+=over 4
+
+=item B<-port>
+
+Act as CMP HTTP server mock-up listening on the given port.
+
+=item B<-max_msgs>
+
+Maximum number of CMP (request) messages the CMP HTTP server mock-up
+should handle, which must be non-negative.
+The default value is 0, which means that no limit is imposed.
+In any case the mock server terminates on internal errors, but not when it
+detects a CMP-level error that it can successfully answer with an error message.
+
+=item B<-srv_ref>
+
+Reference value to use as senderKID of server in case no B<-srv_cert> is given.
+
+=item B<-srv_secret>
+
+Password source for server authentication with a pre-shared key (secret).
+
+=item B<-srv_cert>
+
+Certificate used by the server.
+
+=item B<-srv_key>
+
+Private key used by the server for signing messages.
+
+=item B<-srv_keypass>
+
+Server private key (and cert file) pass phrase source.
+
+=item B<-srv_trusted>
+
+Trusted certificates for client authentication.
+
+=item B<-srv_untrusted>
+
+Intermediate certs for constructing chains for CMP protection by client.
+
+=item B<-rsp_cert>
+
+Certificate to be returned as mock enrollment result.
+
+=item B<-rsp_extracerts>
+
+Extra certificates to be included in mock certification responses.
+
+=item B<-rsp_capubs>
+
+CA certificates to be included in mock ip response.
+
+=item B<-poll_count>
+
+Number oftimes the client must poll before receiving a certificate.
+
+=item B<-check_after>
+
+The checkAfter value (time to wait) to include in poll response.
+
+
+=item B<-grant_implicitconf>
+
+Grant implicit confirmation of newly enrolled certificate.
+
+=item B<-pkistatus>
+
+PKIStatus to be included in server response.
+
+=item B<-failure>
+
+A single failure info bit number to be included in server response.
+
+=item B<-failurebits>
+
+Number representing failure bits to be included in server response.
+
+=item B<-statusstring>
+
+Status string to be included in server response.
+
+=item B<-send_error>
+
+Force server to reply with error message.
+
+=item B<-send_unprotected>
+
+Send response messages without CMP-level protection.
+
+=item B<-send_unprot_err>
+
+In case of negative responses, server shall send unprotected error messages,
+certificate responses (ip/cp/kup), and revocation responses (rp).
+WARNING: This setting leads to behavior violating RFC 4210.
+
+=item B<-accept_unprotected>
+
+Accept missing or invalid protection of requests.
+
+=item B<-accept_unprot_err>
+
+Accept unprotected error messages from client.
+
+=item B<-accept_raverified>
+
+Accept RAVERIFED as proof-of-possession (POPO).
+
+=back
+
+
+=head1 NOTES
+
+When setting up CMP configurations and experimenting with enrollment options
+typically various errors occur until the configuration is correct and complete.
+When the CMP server reports an error the client will by default
+check the protection of the CMP response message.
+Yet some CMP services, in particular those based on EJBCA,
```

Comment:
Probably we shouldn't mention specific products. That kind of information can become out of date very quickly.

Reply:
I've removed the reference to EJBCA.